Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,193 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> closedVolume, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            List<Line> uniqueEdges = UniqueSegments(closedVolume, tolerance);
+
+            if (uniqueEdges.Any(p => point.IsOnCurve(p, tolerance)))
+                return acceptOnEdges;
+            
+            double rayLength = (boundingBox.Max - boundingBox.Min).Length() + 1;
+            Vector rVec;
+            Line line = null;
+            bool isValid = false;
+            int seed = 1;
+            do
+            {
+                rVec = Create.RandomVector(seed).Normalise();
+                isValid = planes.All(p => !rVec.IsInPlane(p, tolerance));
+
+                if (isValid)
+                {
+                    line = new Line { Start = point, End = point + rVec * rayLength };
+                }
+
+                if (!isValid)
+                    seed++;
+            }
+            while (!isValid);
+
+            //Check intersections
+            List<Point> intersectPoints = new List<Point>();
+
+            for (int x = 0; x < planes.Count; x++)
+            {
+                Point intPt = PlaneIntersection(line, planes[x], false, tolerance);
+                if (intPt == null)
+                    continue;
+
+                Polyline pLine = closedVolume[x];
+
+                if  (BH.Engine.Geometry.Query.IsContaining(boundingBox, intPt, true, tolerance) && BH.Engine.Geometry.Query.IsContaining(pLine, new List<Point> { intPt }, false, tolerance))
+                    intersectPoints.Add(intPt);
+            }
+
+            bool isContained = intersectPoints.CullDuplicates().Count % 2 != 0;
+
+            return isContained; //If the number of intersections is odd the point is outsde the space
+        }
+
+        private static List<Line> UniqueSegments(this List<Polyline> outlines, double tolerance)
+        {
+            double sqTol = tolerance * tolerance;
+            List<Line> lines = new List<Line>();
+            foreach (Line line in outlines.SelectMany(x => x.SubParts()))
+            {
+                if (lines.All(x => !x.IsSimilar(line, tolerance)))
```

Comment:
Nothing more fun than correcting own self 😃 
```suggestion
                if (lines.All(x => !x.IsSimilar(line, sqTol)))
```

Reply:
Fixed.